wsi_deid/import_export.py

```diff
@@ -106,14 +107,29 @@ def getSchema():
 
     :returns: an object that can be passed to the jsonschema validator.
     """
-    # TODO: When we have a schema folder, if it exists and any items are in it
-    # that contain a single file that parses as valid json, then we need to
-    # load those files and combine them into a single schema in lieu of using
-    # this default SCHEMA_FILE_PATH
-    logger.info('Using %s (length %d) for schema' % (
-        SCHEMA_FILE_PATH, os.path.getsize(SCHEMA_FILE_PATH)))
-    return json.load(open(SCHEMA_FILE_PATH))
-
+    schemaFolderId = Setting().get(PluginSettings.WSI_DEID_SCHEMA_FOLDER)
+    mergedSchema = {'$schema': 'http://json-schema.org/draft-07/schema','oneOf': []}
+
+    if schemaFolderId:
+        schemaFolder = Folder().load(schemaFolderId, force=True)
+        if schemaFolder:
+            max_files = 1000
+            for item in Folder().childItems(schemaFolder, limit=max_files):
+                if len(list(Item().childFiles(item, limit=2))) == 1:
+                    if len(list(Item().childFiles(item, limit=2))) == 1:
+                        file = next(Item().childFiles(item, limit=1))
+                        if file['size'] < 2e+6:
+                            try:
+                                currentObject = json.load(File().open(file))
+                                if 'properties' in currentObject:
+                                    mergedSchema['oneOf'].append(currentObject)
+                            except Exception:
+                                pass
+        if not len(mergedSchema['oneOf']):
```

Comment:
You want to outdent from this `if` to the bottom of the function so that it returns a schema if there is no folder.

Reply:
fixed in 5d6664ed43a6060a973c354b50ba83ee3e4c49d5